.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs

```diff
@@ -189,6 +190,32 @@ private List<Token> SingleTimePointWithPatterns(string text, DateObject referenc
                     string beforeString = text.Substring(0, (int)extractionResult.Start);
                     ret.AddRange(GetTokenForRegexMatching(beforeString, config.WeekOfRegex, extractionResult));
                     ret.AddRange(GetTokenForRegexMatching(beforeString, config.MonthOfRegex, extractionResult));
+
+                    // Cases like "2 days from/before today/tomorrow"
+                    var isAgo = this.config.AgoRegex.Match(extractionResult.Text).Success;
```

Comment:
Use AgoRegex (containing "before today") and LaterRegex (containing "from today") to judge whether it's cases like "2 days from/before today/tomorrow/yesterday").
If it falls into this category, judge whether it contains "more than" / "less than" / "within" prefix and added into the extract result